code/implementation/host-stats/src/main/java/io/cattle/platform/host/stats/api/ContainerStatsLinkHandler.java

```diff
@@ -0,0 +1,117 @@
+package io.cattle.platform.host.stats.api;
+
+import io.cattle.platform.api.link.LinkHandler;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.constants.HostConstants;
+import io.cattle.platform.core.constants.InstanceConstants;
+import io.cattle.platform.core.model.Host;
+import io.cattle.platform.core.model.Instance;
+import io.cattle.platform.docker.util.DockerUtils;
+import io.cattle.platform.host.api.HostApiUtils;
+import io.cattle.platform.host.model.HostApiAccess;
+import io.cattle.platform.host.service.HostApiService;
+import io.cattle.platform.host.stats.utils.StatsConstants;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class ContainerStatsLinkHandler implements LinkHandler {
+
+    private static final DynamicStringProperty HOST_STATS_PATH = ArchaiusUtil.getString("link.containerstats.path");
+
+    HostApiService hostApiService;
+    ObjectManager objectManager;
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { InstanceConstants.TYPE_CONTAINER, HostConstants.TYPE };
+    }
+
+    @Override
+    public boolean handles(String type, String id, String link, ApiRequest request) {
+        return StatsConstants.CONTAINER_STATS.equals(link);
+    }
+
+    @Override
+    public Object link(String name, Object obj, ApiRequest request) throws IOException {
+        Host host = null;
+        Instance instance = null;
+
+        if (obj instanceof Instance) {
+            instance = (Instance) obj;
+            host = DockerUtils.getHostFromContainer(objectManager, instance);
+        } else if (obj instanceof Host) {
+            host = (Host) obj;
+        }
+
+        if (host == null) {
+            return null;
+        }
+
+        if (!"active".equals(host.getState())) {
+            return null;
+        }
+
+        Map<String, Object> payload = new HashMap<String, Object>();
+        Map<String, Object> containerIdsMap = new HashMap<String, Object>();
+
+        List<Instance> instances = objectManager.mappedChildren(host, Instance.class);
+        for (Instance i : instances) {
```

Comment:
This handler is used for the containerStats link on ***container*** too, right?
The token you are building always has ***all*** containers on the associated host, even if this link is in the context of a a single container.
Can you change this so that it doesn't return all containers when the link is coming from an instance? And can you add a test to explicit confirm this?